portfolio/src/main/java/com/google/sps/servlets/MarkerServlet.java

```diff
@@ -99,4 +125,79 @@ public void storeMarker(Marker marker) {
 
     datastore.put(markerEntity);
   }
+
+  public boolean repeatMarkers(double reportLat, double reportLng, String reportDate,
+   String reportTime, String reportCrime) {
+    Query query = new Query(ENTITY_TITLE);
+    PreparedQuery results = datastore.prepare(query);
+
+    for (Entity entity : results.asIterable()) {
+      double lat = (double) entity.getProperty("lat");
+      double lng = (double) entity.getProperty("lng");
+      String crime = (String) entity.getProperty("crimeType");
+      String date = (String) entity.getProperty("date");
```

Comment:
Not sure why date/time is stored separately and they are string.
You could use date data type, or probably a compact integer data type.